core-java-modules/core-java-concurrency-advanced-4/src/main/java/com/baeldung/producerconsumer/ProducerConsumerDemonstrator.java

```diff
@@ -42,12 +42,14 @@ public static void demoMultipleProducersAndMultipleConsumers() {
         Producer producer = new Producer(dataQueue);
         for(int i = 0; i < producerCount; i++) {
             Thread producerThread = new Thread(producer);
+            producerThread.setName("ProducerThread-" + i);
             producerThread.start();
             threads.add(producerThread);
         }
         Consumer consumer = new Consumer(dataQueue);
         for(int i = 0; i < consumerCount; i++) {
             Thread consumerThread = new Thread(consumer);
+            consumerThread.setName("ConsumerThread-" + i);
```

Comment:
Rollback any changes used for testing requester's claim

Reply:
Done